File: UHFSampleApp/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using UHFAPI_SPC;
using System.Collections;

namespace WindowsTestApp
{
    public partial class Form2 : Form
    {
        SPC_RFID _SPC_RFID = null;
        public Form2()
        {
            InitializeComponent();
            _SPC_RFID = SPC_RFID.getInstance();
        }

        private void btnScan_Click(object sender, EventArgs e)
        {
            _SPC_RFID.tagsDelegate = tagAdd;
            _SPC_RFID.deviceStatusDelegate = devStAdd;
            _SPC_RFID.errorDelegate = errorAdd;
            _SPC_RFID.scanCompleteDelegate = scanComplted;

            if (btnScan.Text.Equals("Start Scan"))
            {
                _SPC_RFID.startScan(false);
                btnScan.Text = "Stop Scan";
            }
            else
            {
                _SPC_RFID.stopScan();
                btnScan.Text = "Start Scan";
            }
            
        }

        public void tagAdd(string a, string r)
        {
            this.Invoke((MethodInvoker)delegate
            {
                a = (!string.IsNullOrWhiteSpace(txtTagsList.Text)) ? "\r\n" + a : a;
                txtTagsList.AppendText(a);
                lblCount.Text = "Count : " + _SPC_RFID.Tags_id.Count;
                txtTagsList.ScrollToCaret();

            });
        }


        public void scanComplted(ArrayList tag)
        {
            this.Invoke((MethodInvoker)delegate
            {
                //lblMessage.Text = tag.ToArray()[0]+" : " + weight+ " : " + user;
                //lstControls.Items.Clear();
                //lstControls.Items.AddRange(tag.ToArray());                
            });
        }


        public void errorAdd(string a)
        {
            this.Invoke((MethodInvoker)delegate
            {
                //lblError.Text = a;
            });

        }

        public void devStAdd(string a)
        {
            this.Invoke((MethodInvoker)delegate
            {
                lblConnection.Text = a;
            });
        }

        private void txtTagsList_TextChanged(object sender, EventArgs e)
        {

        }

        private void lblCount_Click(object sender, EventArgs e)
        {

        }
    }
}

File: UHFSampleApp/VB/WindowsVBTestApp/Form2.vb
Imports UHFAPI_SPC

Public Class Form2
    Private _SPC_RFID As SPC_RFID = Nothing

    Private Sub btnScan_Click(sender As System.Object, e As System.EventArgs) Handles btnScan.Click
        _SPC_RFID.tagsDelegate = AddressOf tagAdd
        _SPC_RFID.deviceStatusDelegate = AddressOf devStAdd
        _SPC_RFID.errorDelegate = AddressOf errorAdd
        _SPC_RFID.scanCompleteDelegate = AddressOf scanComplted

        If btnScan.Text.Equals("Start Scan") Then
            _SPC_RFID.startScan(False)
            btnScan.Text = "Stop Scan"
        Else
            _SPC_RFID.stopScan()
            btnScan.Text = "Start Scan"
        End If
    End Sub



    Public Sub New()
        InitializeComponent()
        _SPC_RFID = SPC_RFID.getInstance()
    End Sub



    Public Sub tagAdd(a As String, r As String)
        Me.Invoke(DirectCast(Sub()
                                 a = If((Not String.IsNullOrWhiteSpace(txtTagsList.Text)), Convert.ToString(vbCr & vbLf) & a, a)
                                 txtTagsList.AppendText(a)
                                 lblCount.Text = "Count : " + _SPC_RFID.Tags_id.Count

                                 txtTagsList.ScrollToCaret()

                             End Sub, MethodInvoker))
    End Sub


    Public Sub scanComplted(tag As ArrayList)
        'lblMessage.Text = tag.ToArray()[0]+" : " + weight+ " : " + user;
        'lstControls.Items.Clear();
        'lstControls.Items.AddRange(tag.ToArray());                
        Me.Invoke(DirectCast(Sub()

                             End Sub, MethodInvoker))
    End Sub


    Public Sub errorAdd(a As String)
        'lblError.Text = a;
        Me.Invoke(DirectCast(Sub()

                             End Sub, MethodInvoker))

    End Sub

    Public Sub devStAdd(a As String)

    End Sub
End Class